internal/api/auth.go

```diff
@@ -87,6 +94,11 @@ func Signup(s storage.Store) http.HandlerFunc {
 
 		go app.SendMail([]string{viper.GetString("email.admin")}, subject, body)
 
+		confirmationBody := "Last step for use Passwall\n\n"
+		confirmationBody += "Confirmation link: " + "http://" + viper.GetString("server.domain") + ":" + viper.GetString("server.port")
+		confirmationBody += "/auth/confirm/" + userDTO.Email + "/" + confirmationCode
+
+		go app.SendMail([]string{userDTO.Email}, "Passwall Email Confirmation", confirmationBody)
```

Comment:
Do you think that there will be no race condition at all ? If so, how can you be sure about it ? 

In my opinion, there is a possibility that race condition might happen. (Two go routines can race between each other, to prevent such a situation generally `waitGroups` are used)

Reply:
I think not because this functions don't have any common variable or access same reader/writer.

Reply:
Oh, I see, my bad, I thought that go routine at line 95 and 101 has common things, the first one just a notification to admin.